--- Code.f	2020-06-25 15:33:58.000000000 -0400
+++ MagmodelinputONE_corrected.f	2013-02-01 15:13:46.000000000 -0500
@@ -2,91 +2,63 @@
 c     This program is to calculate the input parameters for all the Tsyganenko   c
 c     magnetic field models,including G parameters for T01 and T01s, W           c
 c     parameters for TS05.                                                       c
-c     By Zhengrui Qin and Richard Denton, 09/01/2007                             c
-c     Original code from                                                         c
-c     http://virbo.org/svn/virbo/qindenton/                                      c
-c           MagParameterProgram-rsw/MagmodelinputONE_corrected.f                 c
-c     LANL updated version, 2015 (fix for more recent compilers in 2017)         c
+c                                                                                c
+c     By Zhengrui Qin and Richard Denton, 09/01/2007                             c 
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+C      character yesno*3
+      common istart,wst(6)
 cccccc   Choose data format of the OMNIWEB data(http://omniweb.gsfc.nasa.gov/) ccc 
-c      write(*,*)'c    IFORMAT = 1 for one minute data                 c'
-c      write(*,*)'c    IFORMAT = 5 for five minute data                c'
-c      write(*,*)'c    IFORMAT = 60 for hourly data                    c'
-c      write(*,*)'cccccccccccccccccccccccccccccccccccccccccccccccccccccc'
-
-      character ychar*4,fchar*24,filein*32,dumchar*19
-      real dmy(100)
-      common /years/year0,nyrs
-      common wst(6)
-      integer iyear
-
-      write (*,*) ' please input cadence, starting year, no of years'
-      read (*,*) iformat,year0,nyrs
-      write (*,111) iformat,year0,year0+nyrs-1
- 111  format (' input: cadence',i4,' min  starting year',f6.0,
-     & ' ending year',f6.0)
-
+C      write(*,*)''
+C      write(*,*)'Please read readme.txt (where to download input files)'
+C      write(*,*)''
+C      write(*,*)'cccccccccccccccccccccccccccccccccccccccccccccccccccccc' 
+C      write(*,*)'c    Please choose the data format of input files    c'
+C      write(*,*)'c    IFORMAT = 1 for one minute data                 c'
+C      write(*,*)'c    IFORMAT = 5 for five minute data                c'
+C      write(*,*)'c    IFORMAT = 60 for hourly data                    c'
+C      write(*,*)'c    Please input IFORMAT:                           c'
+C      write(*,*)'cccccccccccccccccccccccccccccccccccccccccccccccccccccc'
+C 71   read(*,*)iformat
+      READ *, iformat
+c      write(*,*)'IFORMAT =',iformat
       if(iformat.ne.1.and.iformat.ne.5.and.iformat.ne.60) then
-       write(*,*)' cadence must be 1 or 5 or 60; run aborted'
-       stop
+           write(*,*)'  Input error, please input again.  '
+C           goto 71
       endif
-      if (iformat.eq.1.or.iformat.eq.5) then
-       if (year0.lt.1995) then
-        write (*,*)' starting year must be after 1994; run aborted'
-        stop
-       end if
-      end if
-      if (iformat.eq.60) then
-       if (year0.lt.1963) then
-        write (*,*)' starting year must be after 1962; run aborted' 
-        stop
-       end if
-      end if
-c     if (year0+nyrs.gt.2016) then
-c      write (*,*)' last year cannot be after 2015; run aborted'
-c      stop
-c     end if
-
-      iyear=year0-1
-      write (ychar,'(i4)') iyear
-      fchar=ychar//'/QinDenton_'//ychar//'1231_'
-      filein=fchar//'1min.txt'
-      write (*,*) 'searching ',filein
-      open (unit=1,file=filein,err=20,status='old')       ! search for W1-6 pars in previous file
-      do j=1,200
-       read (1,*,end=20)                                  ! skip header (192 lines)
-      end do
- 90   read (1,*,err=20,end=30) dumchar,(dmy(i),i=1,31),(wst(j),j=1,6)
-       go to 90
- 20   filein=fchar//'5min.txt'
-      write (*,*) 'searching ',filein
-      open (unit=2,file=filein,err=21,status='old')
-      do j=1,200
-       read (2,*,end=21)                  
-      end do
- 91   read (2,*,err=21,end=30) dumchar,(dmy(i),i=1,31),(wst(j),j=1,6)
-       go to 91
- 21   filein=fchar//'hour.txt'
-      write (*,*) 'searching ',filein
-      open (unit=3,file=filein,err=22,status='old')
-      do j=1,200
-       read (3,*,end=22)                  
-      end do
- 92   read (3,*,err=22,end=30) dumchar,(dmy(i),i=1,31),(wst(j),j=1,6)
-       go to 92
- 22   write (*,*) ' did not find previous WG file, will use averages'
-      wst(1) = 0.44    ! at the average values for status 2 in table 3
-      wst(2) = 0.42    ! of Qin,Denton,Tsyganenko,Wolf article
-      wst(3) = 0.66    ! in Space Weather, vol 5, issue 11
-      wst(4) = 0.48    ! http://onlinelibrary.wiley.com/doi/10.1029/2006SW000296/full
-      wst(5) = 0.49   
-      wst(6) = 0.91
-
- 30   write (*,130) wst(1),wst(2),wst(3),wst(4),wst(5),wst(6)
- 130  format ('  initial Ws :',6f7.3)
+C      write(*,*)''
 
+ccccc Starting options ccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+C      write(*,*)'cccccccccccccccccccccccccccccccccccccccccccccccccccccc'
+C      write(*,*)'c    W values starting options                       c'                            
+C      write(*,*)'c    1 to start from 0 for all W parameters          c'
+C      write(*,*)'c    2 to start from user specified values           c'
+C      write(*,*)'c      (W values right before the first record       c'
+C      write(*,*)'c       of the file which you are analyzing)         c' 
+C      write(*,*)'c    e.g., for 5 minute data, enter the W parameters c'
+C      write(*,*)'c      for the preceding 5 minute interval           c'
+C      write(*,*)'c    Input the starting option:                      c'
+C      write(*,*)'c         ( 2 is recommended )                       c'
+C      write(*,*)'cccccccccccccccccccccccccccccccccccccccccccccccccccccc'
+C 73   read(*,*)istart
+C      if(istart.ne.1.and.istart.ne.2) then
+C           write(*,*)'  Input error, please input again.  '
+C           goto 73
+C      endif
+C      if(istart.eq.2) then 
+C      write(*,*)'     Type the starting values for all six W parameters'
+C      write(*,*)'     Six number separated by comma or space'
+C      read(*,*)wst(1),wst(2),wst(3),wst(4),wst(5),wst(6) 
+C      endif
+C      write(*,*)''
+      istart = 2
+      wst(1) = 0.
+      wst(2) = 0.
+      wst(3) = 0.
+      wst(4) = 0.
+      wst(5) = 0.
+      wst(6) = 0.
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
+C      iformat = 60
       if(iformat.eq.1) then
               call kp3_min
               call kpdst_min
@@ -114,32 +86,23 @@
       write(*,*)'     ALL DONE'
       end
 
-cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
-
-c  *************  FOR 1 MIN DATA ***************************************
 
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 cccccc subroutine kp3  ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
 c This program calculates the 3 day average of Kp: Kp3.
-
       subroutine  kp3_min
+      common istart,wst(6)
       integer dst,cap
       character finput*30,fout*30
       real Kp,akp3,Kpa(120),akp,weight,t3day 
-      common wst(6)
-      common /years/year0,nyrs
            
       finput='1min/kpdst.lst' 
       fout='1min/kpkp3dst.d' 
  
       open(unit=1,file=finput,err=33,status='old') 
       goto 34
-
- 33   write(*,*) finput//' NOT EXIST'
+ 33   write(*,*)finput//' NOT EXIST'
       goto 1000
-
  34   open(unit=2,file=fout,status='unknown') 
  
       WRITE(2,*)'YEAR DOY HR   Kp     akp3  DST' 
@@ -150,8 +113,6 @@
       J=0      
 
 10    read(1,*,err=20,end=20) iyr4,idoy,ihr,Kp,dst
-       IF (iyr4.lt.year0.or.iyr4.ge.year0+nyrs) go to 10  ! Alin
-c       IF (iyr4.lt.year0) go to 10  ! Alin
       J=J+1
       if(Kp.eq.99) then
          akp3=99.
@@ -190,12 +151,10 @@
       END  
 cccccccccccccc end of subroutine kp3_min  ccccccccccccccccccccccccccccccccccccccccc
 
-
 ccccccccccccc subroutine kpdstmin   ccccccccccccccccccccccccccccccccccccccccccccccc
-
 c This program interpolate the hourly kp,kp3 and dst to 1 min data.
       subroutine kpdst_min
-      common wst(6)
+      common istart,wst(6)
       character finput*30,fout*30
       real kp1,kp31,kp2,kp32,kp,kp3
       integer dst1,dst2,dst 
@@ -275,15 +234,14 @@
 
 
 cccccccccccc subroutine bsgamma_min  ccccccccccccccccccccccccccccccccccccccccccc
+c     this program adds bs**gamma columns
+        subroutine bsgamma_min
+        common istart,wst(6)
+        dimension bsg(6),gamma(6),idmy(7),dummy(24)
+        character finput*30,fout*30
+        real N,V,P
+	data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
 
-c     this program adds bs**gamma columns    ! runs 10mins for 20years/data, and 5mins for 1year
-      subroutine bsgamma_min
-      dimension bsg(6),gamma(6),idmy(7),dummy(24)
-      character finput*30,fout*30
-      real N,V,P
-      data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
-      common wst(6)
-      common /years/year0,nyrs
     
       finput='1min/omni_min.asc' 
       fout='1min/model_inputr.d' 
@@ -296,15 +254,16 @@
       WRITE(2,400) 'YEAR',' DOY','HR','MIN','ByIMF','BzIMF','V_SW',
      $'Den_P','Pdyn','Bs1g','Bs2g','Bs3g','Bs4g','Bs5g','Bs6g'
 400   format(a5,a4,2a3,3a8,a7,a6,6a8)	
-
 10    read(1,40,err=10,end=1000) iyr,idoy,ihr,imin,(idmy(i),i=1,7),d,i1 
      $,(dummy(i),i=1,24) 
-       IF (iyr.lt.year0.or.iyr.ge.year0+nyrs) go to 10  ! Alin
+40    format(2I4,4I3,3I4,2I7,F6.2,I7, 8F8.2,4F8.1,F7.2, 
+     $F9.0,F6.2,2F7.2,F6.1,6F8.2) 
          By=dummy(5) 
          Bz=dummy(6)
          N=dummy(13)
          V=dummy(9)
          P=dummy(15)  
+         
          if(Bz.eq.9999.99) then
             do i=1,6
                bsg(i)=9999.99
@@ -318,38 +277,29 @@
                bsg(i)=0.0
             enddo
          endif
-40    format(2I4,4I3,3I4,2I7,F6.2,I7, 8F8.2,4F8.1,F7.2, 
-     $F9.0,F6.2,2F7.2,F6.1,6F8.2) 
 
 	write(2,500)iyr,idoy,ihr,imin,By,Bz,V,N,P,(bsg(j),j=1,6)
 	goto 10
 500   FORMAT(I5,i4,2i3,2F8.2,F8.1,F7.2,F6.2,6F8.2)
- 
- 
 	 
 1000  write(*,*)'     Subroutine bsgamma_min    DONE'
-      close(1)
-      close(2)
       end
 cccccccccccccccc  end of bsgamma_min  cccccccccccccccccccccccccccccccccccccccc
 
 
 cccccccccccccccc subroutine av1day  cccccccccccccccccccccccccccccccccccccccccc
-
 c    This program is to take the 1 day average of all parameters.
-
       subroutine av1day_min
-
-      parameter (nipyr=365,nii=30000,nj=11)
-      dimension y(nii),p(nii,nj),w(nii,nj),pv(nj)
+      common istart,wst(6)
+      parameter (nipyr=365,nyr=20,ni=nipyr*nyr, nj=11)
+      dimension y(ni),p(ni,nj),w(ni,nj),pv(nj)
       dimension idmy(7),dummy(24),dd(11),igap(3),it1(3),it2(3)
       double precision yrdouble
       character finput*30,fout*30
-      common wst(6)
-      common /years/year0,nyrs
+           
  
-      ni=nipyr*nyrs
-      iy1= nint(year0)
+      y1= 1995.
+      iy1= nint(y1)
       dyr= 1./nipyr
       do  i= 1,ni
          y(i)= (i-0.5)*dyr
@@ -426,31 +376,34 @@
 
 
 ccccccccccccccc subroutine inter1day cccccccccccccccccccccccccccccccccc
-
 c     This program is to linearly interpolate across the gaps in the data,calculate bz1-bz6
-
       subroutine inter1day_min
-
-      parameter (nk=12)     
+      common istart,wst(6)
+      parameter (ni=365*20,nk=12)     
       character finput*30,fout*30
-      dimension av1(nk,36500),gamma(6)
+      dimension av1(nk,ni),gamma(6)
       data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
-      common wst(6)
-      common /years/year0,nyrs
+     
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+      
+     
 
-      ni=365*nyrs
       finput='1min/av1day.d'
       fout='1min/interav1day.d'
       
-cccc  input the 1 day average
+      
+
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
+ccccccc  input the 1 day average
       open(unit=1,file=finput,status='old')
-      do i=1,ni
-      read(1,*,end=35) (av1(j,i),j=1,nk)
+      
+      do i=1,365*20
+      read(1,*,end=35) (av1(j,i),j=1,12)
       enddo
  35   close(1,status='delete')
 
       do  j=1,ni
-        do  k=2,nk
+        do  k=2,12
         if(av1(k,j).eq.999.) then
             j1=j-1
             j2=j+1
@@ -474,7 +427,7 @@
       enddo
       
       do j=1,ni
-         do k=7,nk
+         do k=7,12
             av1(k,j)=-av1(k,j)**(1.0/gamma(k-6))
          enddo
       enddo
@@ -492,44 +445,44 @@
 
 
 ccccccccccccccc subroutine interpl_min cccccccccccccccccccccccccccccc
-
 c     This program is to interpolate across the gaps in the one-minute data 
-
       subroutine interpl_min
-
-      parameter(nc=11,nd=386,nmax=12999999)            ! nd not used, nmax < 13,100,000
+      common istart,wst(6)
+      parameter(nc=11,nd=386)
       dimension av1(nc+1,99999),istatus(nc),i1(nc),i2(nc), 
      $ainter(nc),check1(nc),check2(nc),check(nc)
-      double precision tyear,datamin(nc+1,nmax)
-      dimension idmy(7),dummy(24),correltime(nc),itime(4,nmax)
+      double precision tyear,datamin(nc+1,9999999)
+      dimension idmy(7),dummy(24),correltime(nc),itime(4,9999999)
       character finput2*30,finput1*30,fout*30
       data correltime/10.59,7.91,64.87,35.95,29.41,
      $7.91,7.91,7.91,7.91,7.91,7.91/
-      common wst(6)
-      common /years/year0,nyrs
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+ 
       
-      finput1='1min/interav1day.d'
+
       finput2='1min/omni_min.asc'
+      finput1='1min/interav1day.d'
       fout='1min/intermin.d'
       n=0
 
-cccc  input the 1 day average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+cccccccc  input the 1 day average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
       open(unit=1,file=finput1,status='old')
+     
       do i=1,99999
       read(1,*,end=34)(av1(j,i),j=1,12)
       enddo
  34   close(1,status='delete')
-
       nday=i-1
-
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
 ccccccc  input the original data   cccccccccccccccccccccccccccccccccccccccccccccccccccccc
 
       open(unit=2,file=finput2,status='old')
-
-      do j=1,nmax
-444   read(2,40,err=20,end=20) iyr4,idoy,ihr,imin,(idmy(i),i=1,7),d,ii
+      do j=1,9999999
+      read(2,40,err=20,end=20) iyr4,idoy,ihr,imin,(idmy(i),i=1,7),d,ii
      $,(dummy(i),i=1,24)
-      IF (iyr4.lt.year0.or.iyr4.ge.year0+nyrs) go to 444  ! Alin
+40    format(2I4,4I3,3I4,2I7,F6.2,I7, 8F8.2,4F8.1,F7.2,
+     $F9.0,F6.2,2F7.2,F6.1,6F8.2)
          n=n+1
          itime(1,n)=iyr4
          itime(2,n)=idoy
@@ -550,9 +503,6 @@
          datamin(12,n)=dummy(6)
       enddo
  20   close(2)
-40    format(2I4,4I3,3I4,2I7,F6.2,I7, 8F8.2,4F8.1,F7.2,
-     $F9.0,F6.2,2F7.2,F6.1,6F8.2)
-
       ni=n
       
 ccccccccccccccccccccc If gap is at the beginning or the end ccccccccccccccc
@@ -591,16 +541,9 @@
      $'V_SW','s','Den_P','s','Pdyn','s','Bz1','s','Bz2','s',
      $'Bz3','s','Bz4','s','Bz5','s','Bz6','s'
 52    format(a5,a4,2a3,2(a8,a2),a8,a2,a7,a2,a6,a2,6(a8,a2))
-      i00=0
-
-      DO 2002 j=1,ni
-
-cw       if (itime(1,j).gt.i00.and.nyrs.gt.1)
-cw     &   write (*,*) ' year',itime(1,j),' started'
-cw       i00=itime(1,j)
-
-       do 4000 k=2,12
 
+      do 2002 j=1,ni
+        do 4000 k=2,12
         n=k-1
         if( ( ( k.eq.2 .or. k.eq.3 .or. k.ge.7 ) 
      $.and. datamin(k,j).ne.9999.99)
@@ -729,16 +672,12 @@
      $ainter(7),istatus(7),ainter(8),istatus(8),ainter(9),istatus(9),
      $ainter(10),istatus(10),ainter(11),istatus(11)
 50    format(i5,i4,2i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,6(f8.2,i2))
-
-2002  Continue
-
+2002	continue					
       close(3)
       write(*,*)'     Subroutine interpl_min    DONE'
       end
-
 cccccccccccccccccc end of interpl_min ccccccccccccccccccccccccccccccc      
 
-
       subroutine year(iyr4,idoy,ihr,imin,year12)
       double precision year12,dum
       if (mod(iyr4,4).eq.0) then
@@ -751,11 +690,11 @@
       end
 
 
-cccccccccccccccccc subroutine WG_min  ccccccccccccccccccccccccccccccccccccc
 
+cccccccccccccccccc subroutine WG_min  ccccccccccccccccccccccccccccccccccccc
 c    This program is to calculate the W and G parameters
       subroutine WG_min
-      common wst(6)
+      common istart,wst(6)
       real kp,kp3
       integer dst
       dimension dat(4,60),idat(4,60),W(6),iW(6),bzn(6),S(6),bsn(6)
@@ -766,54 +705,51 @@
       data xlumda/.39,.46,.39,.42,.41,1.29/
       data r/.39,.7,.031,.58,1.15,.88/
       data f/1.045,1.078,1.017,1.060,1.081,1.280/
-      character finput1*30,finput2*30,fout*30,foutput*28
-      character datechar*10,timechar*8,ychar*4,mchar*2,dchar*2
+      character finput1*30,finput2*30,fout*30
             
-      do i=1,6
-         W(i)=wst(i)/f(i)
-         Wf(i)=0.0
-      enddo
+      if(istart.eq.1) then
+          do i=1,6
+             W(i)=0.0
+          enddo
+      else
+          do i=1,6
+             W(i)=wst(i)/f(i)
+          enddo
+      endif  
               
       pid4 = atan2(1.,1.)
       pi2 = 8*pid4
       n=0
       
+      do i=1,6
+         Wf(i)=0.0
+      enddo
+     
       finput1='1min/intermin.d'
       finput2='1min/kpkp3dstmin.d'
       fout='1min/WGparametersmin.d'  
           
       open(unit=1,file=finput1,status='old')
       open(unit=2,file=finput2,status='old')
-c      open(unit=3,file=fout,status='unknown')
-c      write(3,52) 'Year','Day','Hr','Min','ByIMF','BzIMF','V_SW',
-c     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3','dst'
-c     $,'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6','W1'
-c     $,'W2','W3','W4','W5','W6','6 stat'
-c52    format(a5,a4,a3,a4,a6,7a7,a10,2a7,a6,6a7,6a9,a8)
+      open(unit=3,file=fout,status='unknown')
+      
+      write(3,52) 'Year','Day','Hr','Min','ByIMF','BzIMF','V_SW',
+     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3',
+     $  'dst'
+     $,'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6','W1'
+     $,'W2','W3','W4','W5','W6','6 stat'
+52    format(a5,a4,a3,a4,a6,7a7,a10,2a7,a6,6a7,6a9,a8)
      
       read(1,*)
       read(1,*)
       read(2,*)
-      idayold=0
        
  10   read(2,*,err=20,end=20)iyr1,idy1,ihr1,imn1,kp,kp3,dst
       read(1,70,err=20,end=20)iyr,idy,ihr,imn,by,iby,bz,ibz,v,iv,den,id,
      $p,ip,bz1,i1,bz2,i2,bz3,i3,bz4,i4,bz5,i5,bz6,i6
-       if (iyr1.ne.iyr) write (*,*) 'year mismatch in WGmin input'
  70   format(i5,i4,2i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,6(f8.2,i2))
-
-      call daymonth(iyr,idy,jmo,iday)
-      call utc(iyr,jmo,iday,ihr,imn,datechar,timechar,mchar,dchar)
-      if (idy.ne.idayold) then
-       if (idayold.ne.0) close(3)
-       idayold=idy
-       write (ychar,'(i4)') iyr
-       foutput=ychar//'/QinDenton_'//ychar//mchar//dchar//'_1min'
-       open(unit=3,file=foutput,status='unknown')
-      end if
-
       n=n+1
-
+    
 cccccccc  G parameters cccccccccccccccccccccccccccccccccccc
       if(n.le.60) then
            dat(1,n)=by
@@ -919,54 +855,32 @@
          endif
       enddo
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc     
-
-c      if(idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
-c         write(3,50)iyr,idy,ihr,imn,by,bz,v,den,p,G1,G2,G3,
-c     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bzn(i),i=1,6),
-c     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-c50    format(I5,I4,2I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
-c     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
-c      endif
-      
-      If (idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
-c       if (dst.ne.99999) 
-       write (3,500)
-     &  datechar,timechar,iyr,jmo,iday,ihr,imn,
-     &  by,bz,v,den,p,G1,G2,G3,
-     &  iby,ibz,iv,id,ip,iG1,iG2,iG3,
-     &  kp,kp3,dst,(bzn(i),i=1,6),(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-      End If
-500    format(a10,'T',a8,i5,x,4i3,' 00',x,
-     &  2f7.2,f7.1,x,2f7.2,x,3f7.2,x,
-     &  8i2,x,
-     &  2f6.2,i6,x,6f7.2,x,6f7.3,x,6i2)
-
-      Go To 10
-
+      if(idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
+         write(3,50)iyr,idy,ihr,imn,by,bz,v,den,p,G1,G2,G3,
+     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bzn(i),i=1,6),
+     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
+50    format(I5,I4,2I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
+     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
+      endif
+      
+      goto 10
 20    close(1,status='delete')
       close(2,status='delete')
       close(3)
       write(*,*)'     Subroutine WG_min         DONE'	     
-
       end
 cccccccccccccccccc end of WG_min cccccccccccccccccccccccccccccccccccccccccccc
 
-
-c  **********  FOR 5 MIN DATA ********************
-
-
 cccccc subroutine kp3  cccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
 c This program calculates the 3 day average of Kp: Kp3.
-     
       subroutine  kp3_5min
+      common istart,wst(6)
       integer dst,cap
       character finput*30,fout*30
       real Kp,akp3,Kpa(120),akp,weight,t3day 
-      common wst(6)
-      common /years/year0,nyrs
+     
       
-      finput='5min/kpdst.lst'              ! Kp, Dst data from 1995 to 2015
+      finput='5min/kpdst.lst' 
       fout='5min/kpkp3dst.d' 
  
       open(unit=1,file=finput,err=33,status='old') 
@@ -983,7 +897,6 @@
       J=0      
 
 10    read(1,*,err=20,end=20) iyr4,idoy,ihr,Kp,dst
-       IF (iyr4.lt.year0) go to 10          ! Alin: load data starting at year0
       J=J+1
       if(Kp.eq.99) then
          akp3=99.
@@ -1022,12 +935,10 @@
       END  
 cccccccccccccc end of subroutine kp3_5min  ccccccccccccccccccccccccccccccc
 
-
 cccccccccccccccc kpdst_5min cccccccccccccccccccccccccccccccccccccccccccccc
-
 c    This program interpolate the houly kp and dst to 5-min data
       subroutine kpdst_5min
-      common wst(6)
+      common istart,wst(6)
       character year*4,finput*30,yeard*5,fout*30
       real kp1,kp31,kp2,kp32,kp,kp3
       integer dst1,dst2,dst 
@@ -1102,19 +1013,17 @@
       END 
 cccccccccccccccc end of kpdst_5min ccccccccccccccccccccccccccccccccccccccccccc
 
-
 cccccccccccccccc bsgamma_5min cccccccccccccccccccccccccccccccccccccccccccccccc
-
 c    This program adds the bs**gamma columns
 	subroutine bsgamma_5min
-        common wst(6)
+        common istart,wst(6)
         dimension bsg(6),gamma(6),idmy(7),dummy(24)
         character finput*30,fout*30,aa
         real N,V,P
 	data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
 
       
-      finput='5min/omni_5min.asc'              ! OMNI data from 1981 to 2015
+      finput='5min/omni_5min.asc' 
       fout='5min/model_inputr5min.d' 
  
       open(unit=1,file=finput,err=33,status='old') 
@@ -1148,30 +1057,23 @@
 500   FORMAT(I5,i4,2i3,2F8.2,F8.1,F7.2,F6.2,6F8.2)
 
 1000  continue	  
-      close(1) 
-      close(2) 
       write(*,*)'     Subroutine bsgamma_5min   DONE'
       end
 ccccccccccccccc end of bsgamma_5min   ccccccccccccccccccccccccccccc
 
-
 ccccccccccccccc av1day_5min.f ccccccccccccccccccccccccccccccccccccc
-
 c     This program takes the 1 day average of the data.
-
       subroutine av1day_5min
-
-      parameter (nipyr=365,nii=30000,nj=11)      
-      dimension y(nii),p(nii,nj),w(nii,nj),pv(nj)
+      common istart,wst(6)
+      parameter (nipyr=365,nyr=20,ni=nipyr*nyr, nj=11)
+      dimension y(ni),p(ni,nj),w(ni,nj),pv(nj)
       dimension idmy(7),dummy(24),dd(11),igap(3),it1(3),it2(3)
       double precision yrdouble
       character finput*30,fout*30           
-      common /years/year0,nyrs
-      common wst(6)
  
-      ni=nipyr*nyrs
-      iy1= nint(year0)              ! was 1995, the 1min data over more than 20 years exceeds RAM 
-      dyr= 1./nipyr                   !    also kpdst.lst starts 1995
+      y1= 1995.
+      iy1= nint(y1)
+      dyr= 1./nipyr
       do  i= 1,ni
          y(i)= (i-0.5)*dyr
       enddo
@@ -1184,45 +1086,45 @@
      
       finput='5min/model_inputr5min.d'
       fout='5min/av1day5min.d'
-
       open(unit=11,file=finput,status='old')
-
  34   read(11,20,err=34,end=300)iyr4,idoy,ihr,imin,(pv(j),j=1,nj)
  20   format(I5,i4,2i3,2F8.2,F8.1,F7.2,F6.2,6F8.2)     
        
-      if( mod(iyr4,4).eq.0 ) then
-         doytot= 366.
-      else
-         doytot= 365.
-      endif
+         if( mod(iyr4,4).eq.0 ) then
+            doytot= 366.
+         else
+            doytot= 365.
+         endif
+         iyr4m= iyr4 - iy1
 
-      iyr4m= iyr4 - iy1                                        ! can be negative
-      yv= iyr4m + ( idoy -1. + (ihr + imin/60.0)/24. )/doytot
-      rii= (yv-y(1))/dyr + 1.                                  ! can be negative
-      ii= rii                                                  ! integer(rii)
-      ip= ii + 1           
-      fip= rii - ii
-      fii= 1. - fip
-
-      do 23008 j= 1,nj
-        if( (j.eq.1 .or. j.eq.2).and. pv(j).eq.9999.99 ) go to 23008
-          if( j.eq.3 .and. pv(j).eq.99999.9 ) go to 23008
-          if( j.eq.4 .and. pv(j).eq.999.99 ) go to 23008
-          if( j.eq.5 .and. pv(j).eq.99.99 ) go to 23008
-          if( j.ge.6 .and. pv(j).eq.9999.99) go to 23008
-          if( ii.ge.1 .and. ii.le.ni ) then     ! retains data only between year0 and year0+nyrs
-             p(ii,j)= p(ii,j) + fii*pv(j)
-             w(ii,j)= w(ii,j) + fii
-          endif
-          if( ip.ge.1 .and. ip.le.ni ) then
-             p(ip,j)= p(ip,j) + fip*pv(j)
-             w(ip,j)= w(ip,j) + fip
-          endif
-23008 continue
-      go to 34
+         yv= iyr4m + ( idoy -1. + (ihr + imin/60.0)/24. )/doytot
+
+         rii= (yv-y(1))/dyr + 1.
+         ii= rii
+         ip= ii + 1
+         fip= rii - ii
+         fii= 1. - fip
 
-300   close(11,status='delete')
+         do 23008 j= 1,nj
+
+         if( (j.eq.1 .or. j.eq.2).and. pv(j).eq.9999.99 ) go to 23008
+            if( j.eq.3 .and. pv(j).eq.99999.9 ) go to 23008
+            if( j.eq.4 .and. pv(j).eq.999.99 ) go to 23008
+            if( j.eq.5 .and. pv(j).eq.99.99 ) go to 23008
+            if( j.ge.6 .and. pv(j).eq.9999.99) go to 23008
+            if( ii.ge.1 .and. ii.le.ni ) then
+               p(ii,j)= p(ii,j) + fii*pv(j)
+               w(ii,j)= w(ii,j) + fii
+            endif
+            if( ip.ge.1 .and. ip.le.ni ) then
+               p(ip,j)= p(ip,j) + fip*pv(j)
+               w(ip,j)= w(ip,j) + fip
+            endif
+23008    continue
+
+      go to 34
 
+  300 close(11,status='delete')
       do 23010 j= 1,nj
          do 23012 i= 1,ni
             if( w(i,j).eq. 0. .or.w(i,j).lt.72 ) then
@@ -1232,47 +1134,41 @@
             endif
 23012    continue
 23010 continue
-
       open(unit=21,file=fout,status='unknown')
-      do 23014 i= 1,ni                                 ! output at year0-(year0+nyrs)
-         yrdouble=y(i)+iy1
-         write(21,2100) yrdouble,(p(i,j),j=1,nj)
- 2100    format(f9.4,11(1x,f9.4))
+      do 23014 i= 1,ni
+	   yrdouble=y(i)+iy1
+          write(21,2100) yrdouble,(p(i,j),j=1,nj)
+ 2100       format(f9.4,11(1x,f9.4))
 23014 continue
       close(21)
-
+ 1000 continue 
       write(*,*)'     Subroutine av1day_5min    DONE'
-
       end
 ccccccccccccccccc end of av1day_5min cccccccccccccccccccccccccccccccccccccc
 
-
 ccccccccccccccccc inter1day_5min cccccccccccccccccccccccccccccccccccccccccc
-
 c    This program linearly interpolates across the gaps in the file,calculate bz1-bz6 
-
       subroutine inter1day_5min
-
-      parameter (nk=12)                   
+      common istart,wst(6)
+      parameter (ni=365*20,nk=12)
       character finput*30,fout*30,finput1*30,finput2*30
-      dimension av1(nk,36500),gamma(6)
+      dimension av1(nk,ni),gamma(6)
       data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
-      common /years/year0,nyrs
-      common wst(6)
      
-      ni=365*nyrs
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc   
       finput='5min/av1day5min.d'
       fout='5min/interav1day5min.d'
     
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
 ccccccc  input the 1 day average
       open(unit=1,file=finput,status='old')
-      do i=1,ni
-       read(1,*,end=35) (av1(j,i),j=1,nk)        ! loads data at year0-(year0+nyrs)
+      do i=1,365*20
+      read(1,*,end=35) (av1(j,i),j=1,12)
       enddo
  35   close(1,status='delete')
    
       do  j=1,ni
-        do  k=2,nk
+        do  k=2,12
         if(av1(k,j).eq.999.) then
             j1=j-1
             j2=j+1
@@ -1296,62 +1192,55 @@
       enddo
       
       do j=1,ni
-         do k=7,nk
+         do k=7,12
             av1(k,j)=-av1(k,j)**(1.0/gamma(k-6)) 
          enddo
       enddo
 
       open(unit=2,file=fout,status='unknown')
       do i=1,ni
-         write(2,2100) (av1(j,i),j=1,nk)     ! output at year0-(year0+nyrs)
+         write(2,2100) (av1(j,i),j=1,nk)
  2100    format(f9.4,11(1x,f9.4))
       enddo
       close(2) 
         
       write(*,*)'     Subroutine inter1day_5min DONE'
-
       end
 ccccccccccccccccccc end of inter1day_5min ccccccccccccccccccccccccccccc
 
-
 ccccccccccccccccccc interpl_5min       cccccccccccccccccccccccccccccccc
-
-c    This program  interpolates across the gaps in the 5-minute data     
-
+c    This program  interpolates across the gaps in the one-minute data     
       subroutine interpl_5min
-
-      parameter(nc=11,nmax=5000000)                       ! 105192*nyrs < nmax < 8e6
+      common istart,wst(6)
+      parameter(nc=11)
       dimension av1(nc+1,99999),istatus(nc),i1(nc),i2(nc), 
      $ainter(nc),check1(nc),check2(nc),check(nc)
-      double precision tyear,data5min(nc+1,nmax) 
-      dimension idmy(7),dummy(24),correltime(nc),itime(4,nmax)
+      double precision tyear,data5min(nc+1,2500000)
+      dimension idmy(7),dummy(24),correltime(nc),itime(4,2500000)
       character finput2*30,finput1*30,fout*30,aa
       data correltime/2.76,2.17,14.55,8.37,6.94,
      $2.17,2.17,2.17,2.17,2.17,2.17/
-      common wst(6)
-      common /years/year0,nyrs
-
-      finput1='5min/interav1day5min.d'   
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
       finput2='5min/omni_5min.asc'
+      finput1='5min/interav1day5min.d'
       fout='5min/inter5min.d'
       n=0
-
 cccccccc  input the 1 day average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
       open(unit=1,file=finput1,status='old')
       do i=1,999999
-      read(1,*,end=34)(av1(j,i),j=1,12)    ! loads data at year0-(year0+nyrs)
+      read(1,*,end=34)(av1(j,i),j=1,12)
       enddo
  34   close(1,status='delete')
       nday=i-1
       
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
 ccccccc  input the original data   cccccccccccccccccccccccccccccccccccccccccccccccccccccc
       
       open(unit=2,file=finput2,status='old')
-
-      do i=1,nmax
-555   read(2,40,err=20,end=20) iyr4,idoy,ihr,imin,aa,By,Bz,aa,
+      do i=1,9999999
+      read(2,40,err=20,end=20) iyr4,idoy,ihr,imin,aa,By,Bz,aa,
      $V,aa,den,aa,P,aa
-       IF (iyr4.lt.year0.or.iyr4.ge.year0+nyrs) go to 555  ! Alin: load data  at year0-(year0+nyrs)
+40    format(2I4,2I3,a77,2f8.2,a16,f8.1,a24,f7.2,a9,f6.2,a68)
          n=n+1
          itime(1,n)=iyr4
          itime(2,n)=idoy
@@ -1361,9 +1250,9 @@
          data5min(1,n)=tyear
          data5min(2,n)=By  !by
          data5min(3,n)=Bz  !bz
-         data5min(4,n)=V   !v 
+         data5min(4,n)=V  !v 
          data5min(5,n)=den !den
-         data5min(6,n)=P   !pdyn
+         data5min(6,n)=P !pdyn
          data5min(7,n)=Bz
          data5min(8,n)=Bz
          data5min(9,n)=Bz
@@ -1372,7 +1261,6 @@
          data5min(12,n)=Bz
       enddo
  20   close(2)
-40    format(2I4,2I3,a77,2f8.2,a16,f8.1,a24,f7.2,a9,f6.2,a68)
       ni=n
      
 ccccccccccccccccccccc If gap is at the beginning or the end ccccccccccccccc
@@ -1409,18 +1297,10 @@
      $'V_SW','s','Den_P','s','Pdyn','s','Bz1','s','Bz2','s',
      $'Bz3','s','Bz4','s','Bz5','s','Bz6','s'
 52    format(a5,a4,2a3,2(a8,a2),a8,a2,a7,a2,a6,a2,6(a8,a2))
-      i00=0
-
-      DO 2002 j=1,ni
-
-cw       if (itime(1,j).gt.i00.and.nyrs.gt.1)
-cw     &    write (*,*) ' year',itime(1,j),' started'
-       i00=itime(1,j)
-
-       do 4000 k=2,12
 
+      do 2002 j=1,ni
+        do 4000 k=2,12
         n=k-1
-
         if( ( ( k.eq.2 .or. k.eq.3 .or. k.ge.7 ) 
      $.and. data5min(k,j).ne.9999.99)
      &.or. ( k.eq.4 .and. data5min(k,j).ne.99999.9 ) .or. ( k
@@ -1437,18 +1317,21 @@
             ainter(n)=data5min(k,j)
             check(n)=0              
        else
+
          if(check(n).eq.0)  then         !when check(n).eq.0, it means a new gap coming up
             check(n)=1       
             i1(n)=j-1
             i2(n)=j+1
+                      
             if( k.eq.2 .or. k.eq.3 .or. k.ge.7 ) then
-              DO WHILE (data5min(k,i1(n)) .eq. 9999.99)
+       	      DO WHILE (data5min(k,i1(n)) .eq. 9999.99)
                  i1(n)=i1(n)-1
               END DO
               DO WHILE (data5min(k,i2(n)) .eq. 9999.99)
                  i2(n)=i2(n)+1
               END DO
             endif
+         
             if( k.eq.4 ) then
               DO WHILE (data5min(k,i1(n)) .eq. 99999.9)
                  i1(n)=i1(n)-1
@@ -1457,6 +1340,7 @@
                  i2(n)=i2(n)+1
               END DO
             endif
+     
             if( k.eq.5 ) then
               DO WHILE (data5min(k,i1(n)) .eq. 999.99)
                  i1(n)=i1(n)-1
@@ -1465,6 +1349,7 @@
                  i2(n)=i2(n)+1
               END DO
             endif
+    
             if( k.eq.6 ) then
               DO WHILE (data5min(k,i1(n)) .eq. 99.99)
                  i1(n)=i1(n)-1
@@ -1473,7 +1358,7 @@
                  i2(n)=i2(n)+1
               END DO
             endif
-        endif  
+	 endif  
 
           timegap=(i2(n)-i1(n)-1)/correltime(n)
            if(timegap .le. 2.0)	then
@@ -1491,7 +1376,7 @@
               else if((j-i1(n)) .lt. correltime(n)) then
                 ainter(n)=data5min(k,i1(n))
               else
-                ainter(n)=data5min(k,i2(n))
+	        ainter(n)=data5min(k,i2(n))
               endif
               istatus(n)=1
            endif
@@ -1499,14 +1384,14 @@
            if(timegap .gt. 4.0)	then
               ii=int(correltime(n))+1
               if((j-i1(n)) .lt. ii) ainter(n)=data5min(k,i1(n))
-              if((i2(n)-j) .lt. ii) ainter(n)=data5min(k,i2(n))
+       	      if((i2(n)-j) .lt. ii) ainter(n)=data5min(k,i2(n))
               istatus(n)=1
-              if((j-i1(n)) .ge. ii .and. (i2(n)-j) .ge. ii) then
+       	      if((j-i1(n)) .ge. ii .and. (i2(n)-j) .ge. ii) then
                w1=max(1-(j-i1(n)-correltime(n))/(2.0*correltime(n)),0.0)
                w2=max(1-(i2(n)-correltime(n)-j)/(2.0*correltime(n)),0.0)
-                 wav=1-w1-w2
+    	         wav=1-w1-w2
                  fav=0.0
-                 if(data5min(1,j).lt. av1(1,1)) fav=av1(k,1)
+	         if(data5min(1,j).lt. av1(1,1)) fav=av1(k,1)
                  if(data5min(1,j).gt. av1(1,nday)) fav=av1(k,nday)
                  do  i3=1,99999
                    if( data5min(1,j).ge.av1(1,i3) .and.data5min(1,j).le.
@@ -1515,7 +1400,7 @@
      $-av1(k,i3))/(av1(1,i3+1)-av1(1,i3))+av1(k,i3)
                      goto 35
                     endif
-                 enddo                 
+                 enddo	                  
  35          ainter(n)=w1*data5min(k,i1(n))+w2*data5min(k,i2(n))+wav*fav
                   if(ainter(n) .eq. fav)  istatus(n)=0
                endif
@@ -1530,7 +1415,7 @@
      $.and.(j-i1(n)).ge.3*correltime(n)) then
              istatus(n)=0
          endif
-          
+	          
       endif
                      
 4000  continue
@@ -1541,22 +1426,15 @@
      $ainter(7),istatus(7),ainter(8),istatus(8),ainter(9),istatus(9),
      $ainter(10),istatus(10),ainter(11),istatus(11)
 50    format(i5,i4,2i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,6(f8.2,i2))
-
-2002  Continue
-
+2002  continue					
       close(3)       
       write(*,*)'     Subroutine interpl_5min   DONE'	
       end
-
 cccccccccccccccccccccc end of interpl_5min ccccccccccccccccccccccccccccccccccccc 
-
-
 cccccccccccccccccccccc WG_5min.f ccccccccccccccccccccccccccccccccccccccccccccccc
-ccc  Alin: Calculation of W pars is recursive, all but W3 converge quickly
-
 c     This program is to calculate the W and G parameters
       subroutine WG_5min
-      common wst(6)
+      common istart,wst(6)
       real kp,kp3
       integer dst
       dimension dat(4,60),idat(4,60),W(6),iW(6),bzn(6),S(6),bsn(6)
@@ -1567,55 +1445,50 @@
       data xlumda/.39,.46,.39,.42,.41,1.29/
       data r/.39,.7,.031,.58,1.15,.88/
       data f/1.049,1.065,1.051,1.061,1.050,1.251/
-      character finput1*30,finput2*30,fout*30,foutput*28
-      character datechar*10,timechar*8,ychar*4,mchar*2,dchar*2
+      character finput1*30,finput2*30,fout*30
      
-      do i=1,6
-        W(i)=wst(i)/f(i)
-        Wf(i)=0.0
-      enddo
+      if(istart.eq.1) then          
+          do i=1,6
+             W(i)=0.0
+          enddo
+      else
+          do i=1,6
+             W(i)=wst(i)/f(i)
+          enddo
+      endif  
 
       pid4 = atan2(1.,1.)
       pi2 = 8*pid4
       n=0
       
+      do i=1,6
+         Wf(i)=0.0
+      enddo
+
+      
       finput1='5min/inter5min.d'
       finput2='5min/kpkp3dst5min.d'
       fout='5min/WGparameters5min.d'      
 
       open(unit=1,file=finput1,status='old')
       open(unit=2,file=finput2,status='old')
-c      open(unit=3,file=fout,status='unknown')
-c      write(3,52) 'Year','Day','Hr','Min','ByIMF','BzIMF','V_SW',
-c     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3','dst',
-c     $'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6',
-c     $'W1','W2','W3','W4','W5','W6','6 stat'
-c52    format(a5,a4,a3,a4,a6,7a7,a10,2a7,a6,6a7,6a9,a8)
-
+      open(unit=3,file=fout,status='unknown')
+      write(3,52) 'Year','Day','Hr','Min','ByIMF','BzIMF','V_SW',
+     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3',
+     $  'dst'
+     $,'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6','W1'
+     $,'W2','W3','W4','W5','W6','6 stat'
+52    format(a5,a4,a3,a4,a6,7a7,a10,2a7,a6,6a7,6a9,a8)
+     
       read(1,*)
       read(1,*)
       read(2,*)
-      idayold=0
        
  10   read(2,*,err=20,end=20)iyr1,idy1,ihr1,imn1,kp,kp3,dst
       read(1,70,err=20,end=20)iyr,idy,ihr,imn,by,iby,bz,ibz,v,iv,den,id,
      $p,ip,bz1,i1,bz2,i2,bz3,i3,bz4,i4,bz5,i5,bz6,i6
-       if (iyr1.ne.iyr) write (*,*) 'year mismatch in WG_5min input',
-     & iyr1, iyr
  70   format(i5,i4,2i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,6(f8.2,i2))
- 
-      call daymonth(iyr,idy,jmo,iday)
-      call utc(iyr,jmo,iday,ihr,imn,datechar,timechar,mchar,dchar)
-      if (idy.ne.idayold) then
-       if (idayold.ne.0) close(3)
-       idayold=idy
-       write (ychar,'(i4)') iyr
-       foutput=ychar//'/QinDenton_'//ychar//mchar//dchar//'_5min'
-       open(unit=3,file=foutput,status='unknown')
-      end if
-
       n=n+1
-
 cccccccc  G parameters cccccccccccccccccccccccccccccccccccc
       if(n.le.12) then
            dat(1,n)=by
@@ -1703,7 +1576,7 @@
          endif
 
          S(i)=(den/5)**xlumda(i)*(v/400)**beta(i)*(bsn(i)/5)**gamma(i)
-         W(i)=r(i)/12.0*S(i)+W(i)*exp(-r(i)/12.0)            ! previous W(i) are used !!!!
+         W(i)=r(i)/12.0*S(i)+W(i)*exp(-r(i)/12.0)
          nf=ibz*iv*id
          if(nf.eq.0) then
                Wf(i)=r(i)/12.0*S(i)*0.0+Wf(i)*exp(-r(i)/12.0)	
@@ -1720,123 +1593,30 @@
          endif
       enddo
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc     
-c Alin: matching of years in input files was not inforced (is now), 
-c resulting in combining KpDst data starting 1995 with OMNI W/G parameters starting 1981
-c day-tag mismatch of leap and non-leap years entries led to some years being skipped 
-
-c      if(idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
-c         write(3,50)iyr,idy,ihr,imn,by,bz,v,den,p,G1,G2,G3,
-c     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bzn(i),i=1,6),
-c     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-c50    format(I5,I4,2I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
-c     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
-c      endif
-
-      If (idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
-c       if (dst.ne.99999) 
-       write (3,500)
-     &  datechar,timechar,iyr,jmo,iday,ihr,imn,            
-     &  by,bz,v,den,p,G1,G2,G3,
-     &  iby,ibz,iv,id,ip,iG1,iG2,iG3,
-     &  kp,kp3,dst,(bzn(i),i=1,6),(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-      End If
-500    format(a10,'T',a8,i5,x,4i3,' 00',x,                
-     &  2f7.2,f7.1,x,2f7.2,x,3f7.2,x,
-     &  8i2,x,
-     &  2f6.2,i6,x,6f7.2,x,6f7.3,x,6i2)
+      if(idy1.eq.idy.and.ihr1.eq.ihr.and.imn1.eq.imn) then
+         write(3,50)iyr,idy,ihr,imn,by,bz,v,den,p,G1,G2,G3,
+     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bzn(i),i=1,6),
+     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
+50    format(I5,I4,2I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
+     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
+      endif
       
-      Go To 10
-
+      goto 10
 20    close(1,status='delete')
       close(2,status='delete')
       close(3)
       write(*,*)'     Subroutine WG_5min        DONE'
       end
 ccccccccccccccccccccccccc end of WG_5min ccccccccccccccccccccccccccccccccccccccc
-
-      Subroutine daymonth(iyear,idy,jmo,iday)
-
-      integer jdm(12)
-      data jdm/31,28,31,30,31,30,31,31,30,31,30,31/
-
-      jdm(2)=28
-      if (mod(iyear,4).eq.0) jdm(2)=29
-      iday=idy
-      j=0
- 10   j=j+1
-      if (iday.le.jdm(j)) then
-       jmo=j
-      else
-       iday=iday-jdm(j)
-       go to 10
-      end if
-
-      return
-      End
-cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
-      Subroutine utc(iyr,jmo,iday,ihr,imn,datechar,timechar,mo,dy)
-
-      character datechar*10,timechar*8
-      character year*4,m*1,mo*2,d*1,dy*2,h*1,hr*2,n*1,nm*2
-
-      write (year,'(i4)') iyr
-
-      if (jmo.lt.10) then
-        write (m,'(i1)') jmo
-        mo='0'//m
-       else
-        write (mo,'(i2)') jmo
-       end if
-
-       if (iday.lt.10) then
-        write (d,'(i1)') iday
-        dy='0'//d
-       else
-        write (dy,'(i2)') iday
-       end if
-
-       if (ihr.lt.10) then
-        write (h,'(i1)') ihr
-        hr='0'//h
-       else
-        write (hr,'(i2)') ihr
-       end if
-
-       if (imn.lt.10) then
-        write (n,'(i1)') imn
-        nm='0'//n
-       else
-        write (nm,'(i2)') imn
-       end if
-
-       datechar=year//'-'//mo//'-'//dy
-       timechar=hr//':'//nm//':00'
-
-      return
-      End
-
-
-cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
-
-c     ******   FOR 1 HOUR DATA  ********
-
-
-ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
-
 c This program calculates the 3 day average of Kp,Kp3,and add bs**gamma columns
-
       subroutine kp3bsgamma
+      common istart,wst(6)
       integer dst,cap
       character finput*30,fout*30,aa
       real Kp,akp3,Kpa(120),akp,weight,t3day 
       real N,V,P
       dimension bsg(6),gamma(6)
       data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
-      common wst(6)
-      common /years/year0,nyrs
       
       finput='hour/omni2_hour.dat' 
       fout='hour/model_input.d' 
@@ -1859,13 +1639,11 @@
 
 10    read(1,300,err=20,end=20) iyr4,idoy,ihr,aa,by,bz,aa,N,
      $V,aa,P,aa,Kkp,aa,dst,aa
-       IF (iyr4.lt.year0.or.iyr4.ge.year0+nyrs) go to 10  ! Alin
       Kp=Kkp
      
       
  300  format(2i4,i3,a61,2f6.1,a39,f6.1,f6.0,a18,f6.2,a59,i3,a4,i6,a63)
          if(Bz.eq.999.9) then
-
             do i=1,6
                bsg(i)=9999.99
             enddo
@@ -1917,22 +1695,19 @@
       END 
 cccccccccccccccccccc end of kp3bsgamma  cccccccccccccccccccccccccccccccccccccc
 
-
 cccccccccc  subroutine av1year ccccccccccccccccccccccccccccccccccccccccccccccc
-
 c    This program is to take the 1 year average of the hourly data
-
       subroutine av1year
-      parameter (nipyr=1,nii=1000, nj=11)
-      dimension y(nii),p(nii,nj),w(nii,nj),pv(nj)
+      common istart,wst(6)
+      parameter (nipyr=1,nyr=50,ni=nipyr*nyr, nj=11)
+      dimension y(ni),p(ni,nj),w(ni,nj),pv(nj)
       dimension idmy(7),dummy(24),dd(11),igap(3),it1(3),it2(3)
       double precision yrdouble
       character finput*30,fout*30
-      common wst(6)
-      common /years/year0,nyrs
+           
  
-      ni=nipyr*nyrs
-      iy1=nint(year0)                       ! was 1963
+      y1=1963.
+      iy1=nint(y1)
       dyr= 1./nipyr
       do  i= 1,ni
          y(i)= (i-0.5)*dyr
@@ -2008,22 +1783,19 @@
       end
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 
-
 cccccccccccccc subroutine av20days ccccccccccccccccccccccccccccccccccccccccccccccccc
-
 c    this program is to take the 20 days average of the hourly data
-
       subroutine av20days
-      parameter (nipyr=18,nii=10000,nj=11)
-      dimension y(nii),p(nii,nj),w(nii,nj),pv(nj)
+      common istart,wst(6)
+      parameter (nipyr=18,nyr=50,ni=nipyr*nyr, nj=11)
+      dimension y(ni),p(ni,nj),w(ni,nj),pv(nj)
       dimension idmy(7),dummy(24),dd(11),igap(3),it1(3),it2(3)
       double precision yrdouble
       character finput*30,fout*30
-      common wst(6)
-      common /years/year0,nyrs
+           
  
-      ni=nipyr*nyrs
-      iy1=nint(year0)                ! was 1963
+      y1=1963.
+      iy1=nint(y1)
       dyr= 1./nipyr
       do  i= 1,ni
          y(i)= (i-0.5)*dyr
@@ -2098,14 +1870,11 @@
       end
 ccccccccccccccc end of av20days   ccccccccccccccccccccccccccccccccccc
 
-
 cccccccccc subroutine inter20days ccccccccccccccccccccccccccccccccccc
-
 c    This program is to interpolate across the gaps in the 20 day average data and
 c    calculate bz1-bz6 
-
       subroutine inter20days
-      common wst(6)
+      common istart,wst(6)
       parameter(nc=11)
       dimension av1y(nc+1,99),istatus(nc),i1(nc),i2(nc), 
      $ainter(nc),check1(nc),check2(nc),check(nc),gamma(6)
@@ -2117,13 +1886,14 @@
      $0.39,0.39,0.38,0.38,0.39,0.38/
       data gamma/0.87,0.67,1.32,1.29,0.69,0.53/
 cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+      
 
       finput2='hour/av20days.d'
       finput1='hour/av1year.d'
       fout='hour/inter20days.d'
       
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 cccccccc  input the 20 day average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
       open(unit=1,file=finput1,status='old')
       do i=1,99
       read(1,*,end=35)(av1y(j,i),j=1,12)
@@ -2131,6 +1901,7 @@
  35   close(1,status='delete')
       nd=i-1
       
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
 ccccccc  input the original data   cccccccccccccccccccccccccccccccccccccccccccccccccccccc
       
       open(unit=2,file=finput2,status='old')
@@ -2261,13 +2032,10 @@
       end
 ccccccccccccccccc  end of inter20days   ccccccccccccccccccccccccccccccccccccccccccc 
 
-
 ccccccccccc subroutine interhour cccccccccccccccccccccccccccccccccccccccccccccccccc
-
 c     This program is to interpolate across the gaps in the hourly data     
-
       subroutine interhour
-      common wst(6)
+      common istart,wst(6)
       parameter(nc=11)
       dimension av20(nc+1,1800),istatus(nc),i1(nc),i2(nc), 
      $ainter(nc),check1(nc),check2(nc),check(nc)
@@ -2278,22 +2046,24 @@
      $,finput3*30,finput4*30
       data correltime/1.46,0.73,13.50,2.18,1.68,
      $0.73,0.73,0.73,0.73,0.73,0.73/ 
+cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
             
+    
       finput2='hour/model_input.d'
       finput1='hour/inter20days.d'
       fout='hour/interhour.d'
       n=0
 
-cccccccc  input the 20 days average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 
+cccccccc  input the 20 days average ccccccccccccccccccccccccccccccccccccccccccccccccccccccc
       open(unit=1,file=finput1,status='old')
       read(1,*)
       do i=1,9999
       read(1,*,end=34)(av20(j,i),j=1,12)
       enddo
  34   close(1,status='delete')
-      nd=i-1  
-
+      nd=i-1
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc	
 ccccccc  input the original data   cccccccccccccccccccccccccccccccccccccccccccccccccccccc
       
       open(unit=2,file=finput2,status='old')
@@ -2323,8 +2093,7 @@
          datahr(12,n)=dummy(3)
       enddo
  20   close(2,status='delete')
-
-      nm=n-1
+      nm=n
       
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
       do n=1,nc
@@ -2362,9 +2131,7 @@
 52    format(a5,a4,a3,2(a8,a2),a8,a2,a7,a2,a6,a2,2a6,a6,6(a8,a2))
 
       do 2002 j=1,nm
-
         do 4000 k=2,12
-
         n=k-1
        if( (( k.eq.2 .or. k.eq.3).and. datahr(k,j).ne.999.9).or.
      &(k.eq.4 .and. datahr(k,j).ne.9999. ) .or. 
@@ -2435,6 +2202,7 @@
             endif
 	 endif  
 
+
         
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc            
                 	   
@@ -2506,75 +2274,71 @@
      $ainter(10),istatus(10),ainter(11),istatus(11)
 50    format(i5,i4,i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,
      $2f6.2,i6,6(f8.2,i2))
-
-2002  Continue
-
+2002	continue					
       close(3)
       write(*,*)'     Subroutine interhour      DONE'
       end
        
-cccccccccccccc end of interhour  ccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
+cccccccccccccc end of intehour  ccccccccccccccccccccccccccccccccccccccccccccccccccccc
 
 ccccccccc subroutine WGhour  cccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-
 c     this program is to calculate the W and G parameters
-
       subroutine WGhour
-      common wst(6)
+      common istart,wst(6)
       real kp,kp3,Na,Nb,Ng,Nw
       integer dst
       dimension W(6),iW(6),bzna(6),S(6,4),bznb(6)
       dimension r(6),Wf(6),gamma(6),beta(6),xlumda(6),f(6),bznw(6)
       character str1*8,str2*12  
-      data f/1.069748,1.024995,1.247411,1.100432,0.9768723,1.134791/   
+      data f/1.069748,1.024995,1.247411,1.100432,0.9768723,
+     $1.134791/   
       data gamma/0.87,0.67,1.32,1.29,0.69,0.53/    
       data beta/.8,.18,2.32,1.25,1.6,2.4/
       data xlumda/.39,.46,.39,.42,.41,1.29/
       data r/.39,.7,.031,.58,1.15,.88/
-      character finput*30,fout*30,aa,foutput*28
-      character datechar*10,timechar*8,ychar*4,mchar*2,dchar*2
+      character finput*30,fout*30,aa
       
-      do i=1,6
-        W(i)=wst(i)/f(i)
-        Wf(i)=0.0
-      enddo
+      
+      
+      if(istart.eq.1) then
+          do i=1,6
+             W(i)=0.0
+          enddo
+      else
+          do i=1,6
+             W(i)=wst(i)/f(i)
+          enddo
+      endif  
        
       pid4 = atan2(1.,1.)
       pi2 = 8*pid4
 
+      
+      do i=1,6
+         Wf(i)=0.0
+      enddo
+
+
       finput='hour/interhour.d'
       fout='hour/WGhour.d'      
 
       open(unit=1,file=finput,status='old')
-c      open(unit=3,file=fout,status='unknown')
-c      write(3,52) 'Year','Day','Hr','ByIMF','BzIMF','V_SW',
-c     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3','dst'
-c     $,'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6','W1'
-c     $,'W2','W3','W4','W5','W6','6 stat'
-c52    format(a5,a4,a3,a7,7a7,a10,2a7,a6,6a7,6a9,a8)
+      open(unit=3,file=fout,status='unknown')
+      write(3,52) 'Year','Day','Hr','ByIMF','BzIMF','V_SW',
+     $'Den_P','Pdyn','G1','G2','G3','8 status','kp','akp3',
+     $  'dst'
+     $,'Bz1','Bz2','Bz3','Bz4','Bz5','Bz6','W1'
+     $,'W2','W3','W4','W5','W6','6 stat'
+52    format(a5,a4,a3,a7,7a7,a10,2a7,a6,6a7,6a9,a8)
      
       read(1,*)
       read(1,*)
       Bya = 999.9
-      idayold=0
-      imn=0
              
  10   read(1,70,err=20,end=20)iyr,idy,ihr,by,iby,bz,ibz,v,iv,den,id,
      $p,ip,kp,kp3,dst,bz1,i1,bz2,i2,bz3,i3,bz4,i4,bz5,i5,bz6,i6
  70   format(i5,i4,i3,2(f8.2,i2),f8.1,i2,f7.2,i2,f6.2,i2,
      $2f6.2,i6,6(f8.2,i2))
-
-      call daymonth(iyr,idy,jmo,iday)
-      call utc(iyr,jmo,iday,ihr,imn,datechar,timechar,mchar,dchar)
-      if (idy.ne.idayold) then
-       if (idayold.ne.0) close(3)
-       idayold=idy
-       write (ychar,'(i4)') iyr
-       foutput=ychar//'/QinDenton_'//ychar//mchar//dchar//'_hour'
-       open(unit=3,file=foutput,status='unknown')
-      end if
-
          n=n+1
          Byb = by
          Bzb = bz
@@ -2598,7 +2362,6 @@
                bzna(5)=bznb(5)
                bzna(6)=bznb(6) 
          endif
-
 cccccccc  G parameters cccccccccccccccccccccccccccccccccccc
          dBy = (Byb-Bya)/12.
          dBz = (Bzb-Bza)/12.
@@ -2686,25 +2449,14 @@
                bzna(5)=bznb(5)
                bzna(6)=bznb(6)
          
-c         write(3,50)iyr,idy,ihr,by,bz,v,den,p,G1,G2,G3,
-c     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bznb(i),i=1,6),
-c     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-c50    format(I5,I4,I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
-c     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
- 
-cx       if (dst.ne.99999) 
-        write (3,500)
-     &  datechar,timechar,iyr,jmo,iday,ihr,imn,
-     &  by,bz,v,den,p,G1,G2,G3,
-     &  iby,ibz,iv,id,ip,iG1,iG2,iG3,
-     &  kp,kp3,dst,(bznb(i),i=1,6),(W(i)*f(i),i=1,6),(iW(i),i=1,6)
-500    format(a10,'T',a8,i5,x,4i3,' 00',x,
-     &  2f7.2,f7.1,x,2f7.2,x,3f7.2,x,
-     &  8i2,x,
-     &  2f6.2,i6,x,6f7.2,x,6f7.3,x,6i2)
+         write(3,50)iyr,idy,ihr,by,bz,v,den,p,G1,G2,G3,
+     $iby,ibz,iv,id,ip,iG1,iG2,iG3,kp,kp3,dst,(bznb(i),i=1,6),
+     $(W(i)*f(i),i=1,6),(iW(i),i=1,6)
+50    format(I5,I4,I3,2f7.2,f7.1,2f7.2,3f7.2,i3,7i1,
+     $f7.2,f7.2,i6,6f7.2,6f9.3,i3,5i1)
       
-      Go to 10
-
+      
+      goto 10
 20    close(1,status='delete')
       close(3)
      
